flintrock/scripts/download-hadoop.py

```diff
@@ -18,16 +18,27 @@
 
 if __name__ == '__main__':
     hadoop_version = sys.argv[1]
-    apache_mirror_url = "http://www.apache.org/dyn/closer.lua/hadoop/common/hadoop-{v}/hadoop-{v}.tar.gz?as_json".format(v=hadoop_version)
+    custom_mirror_url = sys.argv[2]
+
+    if custom_mirror_url:
+        mirror_url = custom_mirror_url.format(v=hadoop_version)
+    else:
+        mirror_url = "http://www.apache.org/dyn/closer.lua/hadoop/common/hadoop-{v}/hadoop-{v}.tar.gz?as_json"\
+            .format(v=hadoop_version)
 
     tries = 0
     while tries < 3:
-        mirror_info = json.loads(urlopen(apache_mirror_url).read().decode('utf-8'))
-        file_url = mirror_info['preferred'] + mirror_info['path_info']
-        file_name = os.path.basename(mirror_info['path_info'])
-        print('Downloading file at:', file_url)
+        if custom_mirror_url:
+            print("Downloading file at:", mirror_url)
+            file_name = os.path.basename(mirror_url)
+            file_path, _ = urlretrieve(url=mirror_url, filename=file_name)
+        else:
+            mirror_info = json.loads(urlopen(mirror_url).read().decode('utf-8'))
```

Comment:
If you mean re-querying `http://www.apache.org/dyn/closer.lua/`, then yes. Sometimes an Apache mirror will give us a bad file, so we need to re-query `closer.lua` to get a different mirror.